frontend/public/storage/components/storage-overview/storage-overview.jsx

```diff
@@ -82,6 +86,15 @@ export class StorageOverview extends React.Component {
       diskStats: {
         ...state.diskStats,
         [key]: response,
+      }
+    }));
+  }
+
+  _setDataResiliencyData(key, response) {
```

Comment:
the code for `_setCephDiskStats`, `_setDataResiliencyData`, etc. is very similar. Can you please make it generic?

Reply:
done